cluster-commands.go

```diff
@@ -0,0 +1,266 @@
+	}
+	eCRAddBytes, err := EncryptData(adm.getSecretKey(), crAddBytes)
+	if err != nil {
+		return err
+	}
+
+	reqData := requestData{
+		relPath: adminAPIPrefix + "/cluster-replicate-add",
+		content: eCRAddBytes,
+	}
+
+	resp, err := adm.executeMethod(ctx, http.MethodPut, reqData)
+	defer closeResponse(resp)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return httpRespToErrorResponse(resp)
+	}
+
+	return nil
+}
+
+// CRInfo - contains cluster replication information.
+type CRInfo struct {
+	Enabled                 bool       `json:"enabled"`
+	Name                    string     `json:"name,omitempty"`
+	Clusters                []PeerInfo `json:"clusters,omitempty"`
+	ServiceAccountAccessKey string     `json:"svcAccAccessKey,omitempty"`
+}
+
+// ClusterReplicateInfo - returns cluster replication information.
+func (adm *AdminClient) ClusterReplicateInfo(ctx context.Context) (info CRInfo, err error) {
+	reqData := requestData{
+		relPath: adminAPIPrefix + "/cluster-replicate-info",
+	}
+
+	resp, err := adm.executeMethod(ctx, http.MethodGet, reqData)
+	defer closeResponse(resp)
+	if err != nil {
+		return info, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return info, httpRespToErrorResponse(resp)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return info, httpRespToErrorResponse(resp)
+	}
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return info, err
+	}
+
+	err = json.Unmarshal(b, &info)
+	return info, err
+}
+
+// CRInternalJoinReq - arg body for CRInternalJoin
+type CRInternalJoinReq struct {
+	SvcAcctAccessKey string              `json:"svcAcctAccessKey"`
+	SvcAcctSecretKey string              `json:"svcAcctSecretKey"`
+	SvcAcctParent    string              `json:"svcAcctParent"`
+	Peers            map[string]PeerInfo `json:"peers"`
+}
+
+// PeerInfo - contains some properties of a cluster peer.
+type PeerInfo struct {
+	Endpoint string `json:"endpoint"`
+	Name     string `json:"name"`
+	// Deployment ID is useful as it is immutable - though endpoint may
+	// change.
+	DeploymentID string `json:"deploymentID"`
+}
+
+// CRInternalJoin - used only by minio server to send CR join requests to peer
+// servers.
+func (adm *AdminClient) CRInternalJoin(ctx context.Context, r CRInternalJoinReq) error {
+	b, err := json.Marshal(r)
+	if err != nil {
+		return err
+	}
+	encBuf, err := EncryptData(adm.getSecretKey(), b)
+	if err != nil {
+		return err
+	}
+
+	reqData := requestData{
+		relPath: adminAPIPrefix + "/cluster-replicate/join",
+		content: encBuf,
+	}
+
+	resp, err := adm.executeMethod(ctx, http.MethodPut, reqData)
+	defer closeResponse(resp)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return httpRespToErrorResponse(resp)
+	}
+
+	return nil
+}
+
+// BktOp represents the bucket operation being requested.
+type BktOp string
+
+// BktOp value constants.
+const (
+	// make bucket and enable versioning
+	MakeWithVerBktOp BktOp = "make-with-ver"
```

Comment:
No harm in using complete string 'make-with-version'

Reply:
👍🏾